### PR TITLE
Fixed `Table` growing beyond its limit ([#2098])

### PR DESCRIPTION
**Pull request by njust**
Closes https://github.com/emilk/egui/issues/2098.

image

I checked the Strip and the Table Demo, they are still working fine:
image

### CODE REVIEW

File: crates/egui_extras/src/layout.rs [hunk 1]
```diff
@@ -101,7 +101,13 @@ impl<'l> StripLayout<'l> {
         let rect = self.cell_rect(&width, &height);
         let used_rect = self.cell(rect, add_contents);
         self.set_pos(rect);
-        self.ui.allocate_rect(rect.union(used_rect), Sense::hover())
+        let rect = if !rect.contains_rect(used_rect) {
+            rect.union(self.ui.available_rect_before_wrap())
+        } else {
+            used_rect
+        };
```
Comment (emilk) [inline crates/egui_extras/src/layout.rs:108]: perhaps you could add a comment here explaining the logic of this

Comment (njust) [reply crates/egui_extras/src/layout.rs:109]: I changed that line to `self.ui.available_rect_before_wrap()` and added a comment.

**Comment by njust** [comment]
I marked this PR as a draft. While the examples in the Demo App are working, I noticed some jitter when permanently adding rows: 
Peek 2022-10-02 15-41

**Comment by njust** [comment]
The latest change, fixes the jitter.

**Comment by njust** [comment]
I need to restructure my fork, so I'm closing this.

**Pull request closed without merging**